modules/fledgeForGpt.js

```diff
@@ -59,111 +39,36 @@ function setComponentAuction(adUnitCode, auctionConfigs) {
   }
 }
 
-function onAuctionInit({auctionId}) {
-  PENDING[auctionId] = {};
-}
-
-function getSlotSignals(bidsReceived = [], bidRequests = []) {
-  let bidfloor, bidfloorcur;
-  if (bidsReceived.length > 0) {
-    const bestBid = bidsReceived.reduce(maximum(currencyCompare(bid => [bid.cpm, bid.currency])));
-    bidfloor = bestBid.cpm;
-    bidfloorcur = bestBid.currency;
-  } else {
-    const floors = bidRequests.map(bid => typeof bid.getFloor === 'function' && bid.getFloor()).filter(f => f);
-    const minFloor = floors.length && floors.reduce(minimum(currencyCompare(floor => [floor.floor, floor.currency])))
-    bidfloor = minFloor?.floor;
-    bidfloorcur = minFloor?.currency;
-  }
-  const cfg = {};
-  if (bidfloor) {
-    deepSetValue(cfg, 'auctionSignals.prebid.bidfloor', bidfloor);
-    bidfloorcur && deepSetValue(cfg, 'auctionSignals.prebid.bidfloorcur', bidfloorcur);
+export function onAuctionConfigFactory(setGptConfig = setComponentAuction) {
+  return function onAuctionConfig(auctionId, adUnitCode, auctionConfig) {
+    if (autoconfig) {
+      setGptConfig(adUnitCode, auctionConfig.componentAuctions);
+    }
   }
-  return cfg;
 }
 
-function onAuctionEnd({auctionId, bidsReceived, bidderRequests}) {
-  try {
-    const allReqs = bidderRequests?.flatMap(br => br.bids);
-    Object.entries(PENDING[auctionId]).forEach(([adUnitCode, auctionConfigs]) => {
-      const forThisAdUnit = (bid) => bid.adUnitCode === adUnitCode;
-      const slotSignals = getSlotSignals(bidsReceived?.filter(forThisAdUnit), allReqs?.filter(forThisAdUnit));
-      setComponentAuction(adUnitCode, auctionConfigs.map(cfg => mergeDeep({}, slotSignals, cfg)))
+export function setPAAPIConfigFactory(
+  getPAAPIConfig = (filters) => getGlobal().getPAAPIConfig(filters),
+  setGptConfig = setComponentAuction) {
+  return function(filters) {
```

Comment:
If `autoconfig` is `true` when this is called, we will see each _"register component auction config"_ log message twice. It is harmless but potentially confusing, or we can skip the execution since it's meant for the other case.
Thoughts ?